src/core/jsonschema/jsonschema_vocabularies.cc

```diff
@@ -0,0 +1,239 @@
+#include <sourcemeta/core/jsonschema_vocabularies.h>
+
+#include <optional> // std::optional
+#include <string>   // std::string
+#include <utility>  // std::pair
+#include <vector>   // std::vector
+
+namespace {
+// Map vocabulary URI to its corresponding enum flag
+auto uri_to_known_vocabulary(std::string_view uri)
+    -> std::optional<sourcemeta::core::KnownVocabulary> {
+  using sourcemeta::core::KnownVocabulary;
+
+  // Pre-vocabulary dialects
+  if (uri == "http://json-schema.org/draft-00/schema#")
```

Comment:
Looks good but can you do it the other way around? Let's prefer matching 2020-12 first, then 2019-09, etc. The reason is that latest vocabularies are used more often, so you can short circuit faster. i.e. almost nobody uses Draft 0 right now, but you would still check for that first